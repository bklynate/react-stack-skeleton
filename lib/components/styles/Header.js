'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactcss = require('reactcss');

var _reactcss2 = _interopRequireDefault(_reactcss);

var _responsiveEnhancements = require('../../containers/responsiveEnhancements');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The header for the website.
 */
function Header(_ref) {
  var screenSize = _ref.screenSize,
      title = _ref.title,
      menu = _ref.menu,
      otherProps = _objectWithoutProperties(_ref, ['screenSize', 'title', 'menu']);

  var styles = (0, _reactcss2.default)({
    'default': {
      header: {
        width: '100%',
        height: 150,
        backgroundColor: "black"
      },
      title: {
        position: 'absolute',
        top: 10,
        left: 10,
        color: 'white'
      },
      menu: {
        position: 'absolute',
        top: 10,
        right: 10
      }
    },
    'screenSize-xs': {
      header: {
        height: 100
      },
      title: {
        top: 5
      },
      menu: {
        top: 5
      }
    }
  }, { screenSize: screenSize });

  return _react2.default.createElement(
    'header',
    _extends({ style: styles.header }, otherProps),
    _react2.default.createElement(
      'h1',
      { style: styles.title },
      title
    ),
    _react2.default.createElement(
      'div',
      { style: styles.menu },
      menu
    )
  );
}
exports.default = (0, _responsiveEnhancements.useScreenSize)(Header);