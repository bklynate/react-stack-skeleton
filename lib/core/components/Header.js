'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Header;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The header for the site.
 */
function Header(_ref) {
  var otherProps = _objectWithoutProperties(_ref, []);

  return _react2.default.createElement(
    'header',
    otherProps,
    'NYI: Header'
  );
}