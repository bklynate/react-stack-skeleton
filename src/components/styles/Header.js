/* @flow */
import React from 'react';
import reactCSS from 'reactcss';

import { useScreenSize } from '../../containers/responsiveEnhancements';

type Props = {
  screenSize: string,
  title: string,
  menu: React.Element<*>,
};

/**
 * The header for the website.
 */
function Header({ screenSize, title, menu, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      header: {
        width: '100%',
        height: 150,
        backgroundColor: "black",
      },
      title: {
        position: 'absolute',
        top: 10,
        left: 10,
        color: 'white',
      },
      menu: {
        position: 'absolute',
        top: 10,
        right: 10,
      },
    },
    'screenSize-xs': {
      header: {
        height: 100,
      },
      title: {
        top: 5,
      },
      menu: {
        top: 5,
      },
    },
  }, { screenSize });

  return (
    <header style={styles.header} {...otherProps}>
      <h1 style={styles.title}>{title}</h1>
      <div style={styles.menu}>{menu}</div>
    </header>
  );
}
export default useScreenSize(Header);