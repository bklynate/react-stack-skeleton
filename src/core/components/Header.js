/* @flow */
import React from 'react';

type Props = {

};

/**
 * The header for the site.
 */
export default function Header({ ...otherProps }: Props): React.Element<*> {
  return (
    <header {...otherProps}>
      NYI: Header
    </header>
  );
}
