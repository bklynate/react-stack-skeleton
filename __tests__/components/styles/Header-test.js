import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../../../src/components/styles/Header';

jest.mock('../../../src/containers/responsiveEnhancements', () => ({
  useScreenSize: jest.fn((C) => (props) => <C screenSize="lg" {...props} />),
}));

test('The Header component displays the given title.', () => {
  const component = renderer.create(
    <Header title="Test Title" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('The Header component displays the given menu component.', () => {
  const component = renderer.create(
    <Header
      menu={<div>Test Menu</div>}
    />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
