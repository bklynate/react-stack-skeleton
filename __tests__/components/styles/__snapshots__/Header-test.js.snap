exports[`test The Header component displays the given menu component. 1`] = `
<header
  style={
    Object {
      "backgroundColor": "black",
      "height": 150,
      "width": "100%",
    }
  }>
  <h1
    style={
      Object {
        "color": "white",
        "left": 10,
        "position": "absolute",
        "top": 10,
      }
    } />
  <div
    style={
      Object {
        "position": "absolute",
        "right": 10,
        "top": 10,
      }
    }>
    <div>
      Test Menu
    </div>
  </div>
</header>
`;

exports[`test The Header component displays the given title. 1`] = `
<header
  style={
    Object {
      "backgroundColor": "black",
      "height": 150,
      "width": "100%",
    }
  }>
  <h1
    style={
      Object {
        "color": "white",
        "left": 10,
        "position": "absolute",
        "top": 10,
      }
    }>
    Test Title
  </h1>
  <div
    style={
      Object {
        "position": "absolute",
        "right": 10,
        "top": 10,
      }
    } />
</header>
`;
